### DISCUSSION TITLE
What Distinguishes Aurora and CodeEdit From Another?

### DISCUSSION THREAD

**Discussion by codeface-io** [AuroraEditor/AuroraEditor#286] [Q&A] [answered]
I just stumbled upon Aurora but have been following CodeEdit for some time.

Both are open source and seem to aim at the same goal.

So why is there this seeming divergence of efforts? Is there shared code under the hood? Are the goals not so similar after all? Is it pure co-incidence or was one project spawned from the other at some point?

I feel this apparent parallel should be addressed up front 😊

**Comment by 0xWDG** [comment, marked as answer]
We all worked on CodeEdit, and some of us still do things for the CE community.

There are a few differences. 
1) we encourage small PR’s which may not be working perfectly (we think this will speed up the development of the project)
2) ~we use a different highlighting engine, because we think tree-sitter is not fast enough (at least for initial parsing)~
   _Update MAY-2024:_ We've switched to tree-sitter, we're still not okay with the speed. we try to optimize it.
3) we removed the massive amount of modules which keeps failing to fetch and needed extra instructions for users
4) we’re focusing more on functionality then appearance, CodeEdit want super nice designs, our priority is to get it working, and then look at design


some of us had the feeling that their ideas where not good enough, some of us suggested things, and the response was no, but a few weeks later it was implemented because an admin had “this” idea

**Reply by codeface-io** [reply]
What a quick response! And yes, that answers my question and I understand where you're coming from ...

**Comment by austincondiff** [comment]
Hi @flowtoolz, I am the organizer of CodeEdit. I am a little confused myself as our project was meant to be a unifying project to rally together awesome developers to create a great native free and open-source code editor for the community. The things that confuse me in response to the above list are as follows:

1. We also encourage small PR's. This ensures quality reviews and decreased risk.
2. Yes, we do believe tree-sitter is faster and more efficient than anything else we have found as it highlights code incrementally rather than wholesale for each and every change in code and it is written lower level than other solutions, however we are certainly open to conversation around this and are constantly working to improve performance.
3. We are right in the middle of a big module restructure to reduce the barrier of entry for new and existing contributors, which is partially inspired by what has been done at Aurora in their own restructure.
4. We also understand that first and foremost things needs to work, but we also want to provide users with the experience they have come to expect on the Mac. Design attention can certainly happen after it is working well and sometimes that is necessary.

@0xWDG I think there is a lot more overlap between our two projects than what is perceived. There were a few conversations that I remember having quite a while ago that may have been taken out of context or in a different way than intended which caused negative feelings and ultimately led to the fork from CodeEdit (later named AuroraEditor).

I certainly welcome forks from our project, that is the beauty of open source. However, where possible, I think we are more powerful together than apart because we are not duplicating effort and fragmenting the project, which confusing potential contributors.

The bottom line is, both projects are producing fantastic work, but I believe there is a much higher probability of success if we worked together. I'd encourage y'all to consider a conversation with us to see if there is any room to unify our efforts. If not, I wish y'all the best of luck!

**Reply by codeface-io** [reply]
Thanks for expanding the view on this!

> I'd encourage y'all to consider a conversation with us to see if there is any room to unify our efforts.

+1 to that. IMHO, what enables complex systems and what allows people and teams to combine their efforts are **interfaces** i.e. reliable agreements:

1. Common versioned API (like an extension-/plugin system)
2. Common versioned building blocks (like Swift packages and command line tools)

How great would it be if both extract and unify some common underlying infrastructure. And how great would it be if plugins/extensions developed for one would also work in the other 😃

I think there's potential on both these frontiers.

All the best to CodeEdit + Aurora! 🚀🚀

**Reply by codeface-io** [reply]
> I think there is a lot more overlap between our two projects than what is perceived.

It seems to me the goals are equal enough, only the philosophy on how to get there might differ. But that leaves all the more room for both teams to put at least their results back together.

**Reply by austincondiff** [reply]
I guess "unify our efforts" is up for interpretation. 😄

**Reply by KaiTheRedNinja** [reply]
I am open to the concept of a unification, however ae did split due to disagreements with the _people_ and not just code related things. That would have to be worked out first.

**Reply by codeface-io** [reply]
> I guess "unify our efforts" is up for interpretation. 😄

I just think it has to be done iteratively.
